extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -392,4 +428,20 @@ public void thingUpdated(Thing thing) {
             }
         }
     }
+
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+        while (getThing().getStatus() == ThingStatus.UNKNOWN || getThing().getStatus() == ThingStatus.INITIALIZING
+                || (getThing().getStatus() != ThingStatus.ONLINE && currentConfigStatus.getConfigStatusMessages().isEmpty())) {
+            try {
+                synchronized (this) {
+                    this.wait();
```

Comment:
You don't want to block the calling thread, do you? 😉

Reply:
Actually - that's exactly what I want

Reply:
without knowing all the details this sounds like it should be handled in an asynchrone way. The UI should idle as long as the desired state is not reached. The backend should not be blocked by this. wdyt?